bundles/org.openhab.binding.modbus.helioseasycontrols/src/main/java/org/openhab/binding/modbus/helioseasycontrols/internal/HeliosEasyControlsHandler.java

```diff
@@ -0,0 +1,807 @@
+                            }
+                        }
+                    } else {
+                        handleError("Variable definition is null", ThingStatusDetail.CONFIGURATION_ERROR);
+                    }
+                }, config.getRefreshInterval(), config.getRefreshInterval(), TimeUnit.MILLISECONDS);
+            }
+        } else { // at least one null assertion has failed, let's log the problem and update the thing status
+            if (this.comms == null) {
+                this.handleError("Modbus communication interface is unavailable",
+                        ThingStatusDetail.COMMUNICATION_ERROR);
+            }
+            if (this.variableMap == null) {
+                this.handleError("Variable definition is unavailable", ThingStatusDetail.CONFIGURATION_ERROR);
+            }
+            if (this.config == null) {
+                this.handleError("Binding configuration is unavailable", ThingStatusDetail.CONFIGURATION_ERROR);
+            }
+        }
+    }
+
+    @Override
+    public void dispose() {
+        if (this.pollingJob != null) {
+            this.pollingJob.cancel(true);
+        }
+        this.comms = null;
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        String channelId = channelUID.getIdWithoutGroup();
+        if (command instanceof RefreshType) {
+            if (channelId.equals(HeliosEasyControlsBindingConstants.SYS_DATE)) {
+                scheduler.submit(() -> readValue(HeliosEasyControlsBindingConstants.DATE));
+                scheduler.submit(() -> readValue(HeliosEasyControlsBindingConstants.TIME));
+            } else if (channelId.equals(HeliosEasyControlsBindingConstants.BYPASS_FROM)) {
+                scheduler.submit(() -> readValue(HeliosEasyControlsBindingConstants.BYPASS_FROM_DAY));
+                scheduler.submit(() -> readValue(HeliosEasyControlsBindingConstants.BYPASS_FROM_MONTH));
+            } else if (channelId.equals(HeliosEasyControlsBindingConstants.BYPASS_TO)) {
+                scheduler.submit(() -> readValue(HeliosEasyControlsBindingConstants.BYPASS_TO_DAY));
+                scheduler.submit(() -> readValue(HeliosEasyControlsBindingConstants.BYPASS_TO_MONTH));
+            } else {
+                scheduler.submit(() -> readValue(channelId));
+            }
+        } else { // write command
+            String value = null;
+            if (command instanceof OnOffType) {
+                value = command == OnOffType.ON ? "1" : "0";
+            } else if (command instanceof DateTimeType) {
+                ZonedDateTime d = ((DateTimeType) command).getZonedDateTime();
+                if (channelId.equals(HeliosEasyControlsBindingConstants.SYS_DATE)) {
+                    setSysDateTime(d);
+                } else if (channelId.equals(HeliosEasyControlsBindingConstants.BYPASS_FROM)) {
+                    this.setBypass(true, d.getDayOfMonth(), d.getMonthValue());

```

Comment:
Should you convert `ZonedDateTime` to helios system time zone before extracting dayOfMonth etc?

Reply:
I don't think this is necessary for setting the bypass' from and to date. These settings are independent from the actual system time.

Reply:
Let me try to make point in a different way 

Surely you agree that 2020-01-02 01:00 UTC and 2020-01-01 23:00 UTC-2h represent exactly same moment in time?

Just extracting the date from "local dates" yields different results.

Reply:
Yes, I'm aware of the GMT offset having an impact on the day in the date. But in this specific case (setting the bypass date) I don't see the relevance.
The user, a rule, etc. wants to change the day from which on the bypass functionality should be active or deactivated. This will result in a date with its day and month based on the local time zone being assigned to an item. Now I grab precisely this day and month and write this to the device.
Maybe I'm missing something, but in this case I'm just using the DateTime type as a vehicle to get month and day set together. Please also see the private class BypassDate for my internal representation which is completely independent of any time zone.

Reply:
I see what you are saying but I think you are missing my point.

I will try describe the whole flow and why I think you should consider taking zones into account 

Given
- user with local time zone UTC -6. User has configured openhab correctly that it uses this timezone. Ui elements etc use UTC -6 
- Helios with UTC timezone 

Now, user open openhab ui and sets bypass date to Jan 02. 

Currently binding writes Jan 02 to Helios. Helios interprets Jan 02 date against its system time of course. So bypass mode gets actually activated Jan 02 00:00 UTC, which corresponds to Jan 01 18:00 in UTC-6.

So as summary user thought he set bypass to Jan 2 (his time) but it is actually Jan 1 (his time)

I might be missing something as well but hopefully the above opens up my thinking

Reply:
I think I get your point. However, if the user used the device's own configuration web interface and set the date there, the result would be the same. In the end the device will use its own system time to activate/deactivate the bypass.

After all, it's more important that the settings in the device according to its local time zone are correct. The bypass depends both on a period of time in which it will be considered as well as on the outside and room temperature. So this is something very local and quite dpendent on the local time as well.

So even if it were a valid use case that I run an OH instance in Austria to control a ventilation system in the US, I would, as a user, still want to set the bypass to a specific date, when I want the bypass to become active in the US, not having to consider possible time zone differences.

I know it's only one line of code, but I'm still not convinced 😉